### PR TITLE
Change extrudegeometry to buffergeometry

### PR DESCRIPTION
**Pull request by gero3**
This makes the extrudegeometry emit a buffergeometry instead of a geometry.

### CODE REVIEW

**Comment by WestLangley** [comment]
Thanks for this!

It appears, however, that `ExtrudeGeometry` does not support an array of materials with this PR.

Also, `ExtrudeBufferGeometry` appears to be missing normals.

File: src/geometries/ExtrudeGeometry.js [hunk 1]
```diff
@@ -94,9 +150,9 @@ ExtrudeGeometry.prototype.addShape = function ( shape, options ) {
 	var uvgen = options.UVGenerator !== undefined ? options.UVGenerator : ExtrudeGeometry.WorldUVGenerator;
 
 	var splineTube, binormal, normal, position2;
-	if ( extrudePath ) {
```
Comment (mrdoob) [inline src/geometries/ExtrudeGeometry.js]: Any chance you can revert the formatting changes?

**Comment by gero3** [comment]
The unnecessary layout changes are reverted and normals + groups are implemented.

**Comment by WestLangley** [comment]
Sweet! Thanks!

Unrelated to this PR, it appears that the `ExtrudeMaterial` options `material` and `extrudeMaterial`  (they identify the material index of the front/back and sidewall materials) were dropped from the API in r.74 and never reinstated. You can see these settings being used in the examples.

Should we reinstate them or remove them?

Also unrelated to this PR, using `DefaultUVGenerator`, the UVs on the back-face are reversed.

I am not sure of a clean fix for this other than to support a separate material (and texture) for the back-face.

**Comment by gero3** [comment]
I made a new PR #10921 to add options material and extrudeMaterial back into the mix.

The UVs is something I might look into later.

**Comment by gero3** [comment]
Is this good enough to add to the source or are there still things that are broken, @WestLangley??

**Comment by WestLangley** [comment]
@gero3 Except for some whitespace issues, your PR looks good to me. Give yourself some attribution! :)

Docs and docs example will need to be updated for `BufferGeometry`, multi-material support, material index support, etc. /ping @looeee

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Hmm... Seems like this example broke though...
http://localhost:8000/examples/webgl_geometry_extrude_shapes.html

**Comment by looeee** [comment]
> Docs and docs example will need to be updated for BufferGeometry, multi-material support, material index support, etc. /ping @looeee

I don't have time to update the docs every time something gets changed, unfortunately . If we are serious about keeping them up to date it really needs to be the responsibility of the person who makes changes to at least make rudimentary documentation updates to go along with it.

**Pull request merged**